packages/hub/src/app/models/[owner]/[slug]/EditSquiggleSnippetModel.tsx

```diff
@@ -111,8 +89,42 @@ export const EditSquiggleSnippetModel: FC<Props> = ({ modelRef }) => {
     };
   }, [content, revision.relativeValuesExports]);
 
-  const form = useForm<FormShape>({
+  const { form, onSubmit, inFlight } = useMutationForm<
+    FormShape,
+    EditSquiggleSnippetModelMutation,
+    "UpdateSquiggleSnippetResult"
```

Comment:
This duplicates the `expectedTypename` field, and unfortunately it's necessary for type safety. (It's also necessary in `useAsyncMutation` that we had before, but this hook's type parameters are stricter).

This is a known TypeScript limitation: it lacks partial type argument inference (see https://github.com/microsoft/TypeScript/issues/26242 for the details).

I hope they'll get around to adding it in the next 6-12 months (people have been complaining about its priority for a long time, e.g. https://github.com/microsoft/TypeScript/issues/55486#issuecomment-1690681098), but until then we'll have to tolerate it.